File: Acity_JAVA_Project/Griselda Nyasembi/JAVA_PROJECT/FileHandler.java
import java.io.*;
import java.util.ArrayList;
import java.util.List;

public class FileHandler extends KeywordSearcher {
    private String filename;

    public void setFilename(String name) {
        this.filename = name;
    }

    public String getFilename() {
        return filename;
    }

    public boolean fileExists() {
        File file = new File(filename);
        return file.exists();
    }

    public List<String> readFile() {
        List<String> lines = new ArrayList<>();
        try (BufferedReader reader = new BufferedReader(new FileReader(filename))) {
            String line;
            while ((line = reader.readLine()) != null) {
                lines.add(line);
            }
        } catch (IOException e) {
            System.err.println("Error reading file: " + e.getMessage());
        }
        return lines;
    }

    public void saveToFile(List<String> lines, String outputFilename) {
        try (BufferedWriter writer = new BufferedWriter(new FileWriter(outputFilename))) {
            for (String line : lines) {
                writer.write(line);
                writer.newLine();
            }
            System.out.println("Results saved to " + outputFilename);
        } catch (IOException e) {
            System.err.println("Error writing to file: " + e.getMessage());
        }
    }
}

File: Acity_CPP_Project/IT/IT_AI_Group15/FileHandler.cpp
#include <iostream>
#include <fstream>
#include <vector>
#include <string>
#include "KeywordSearcher.cpp"
using namespace std;

class FileHandler: public KeywordSearcher{
private:
    string filename;

public:
    void setFilename(string name) {
         filename = name; 
         }

    string getFilename()  { 
        return filename; 
        }

    bool fileExists()  {
        ifstream file(filename);
        return file.is_open();
    }

    vector<string> readFile() {
        ifstream file(filename);
        vector<string> lines;

        string line;
        while (getline(file, line)) {
            lines.push_back(line);
        }
        file.close();
        return lines;
    }

    void saveToFile(vector<string> lines, string outputFilename) {
        ofstream outFile(outputFilename);
        if (!outFile.is_open()) {
            cerr << "Error opening output file." << endl;
            return;
        }
        for (auto line : lines) {
            outFile << line << endl;
        }
        outFile.close();
        cout << "Results saved to " << outputFilename << endl;
    }
};
